src/xpk/core/cluster_private.py

```diff
@@ -51,36 +80,16 @@ def authorize_private_cluster_access_if_necessary(args) -> int:
   )
 
   if not new_authorized_networks_needed:
-    return_code, current_machine_already_authorized = (
-        is_current_machine_in_any_network(authorized_networks)
-    )
-    if return_code > 0:
+    (
+        is_current_machine_in_any_network_return_code,
+        current_machine_already_authorized,
+    ) = is_current_machine_in_any_network(authorized_networks)
+    if is_current_machine_in_any_network_return_code > 0:
       xpk_print("Error checking current machine's IP address.")
-      return return_code
+      return is_current_machine_in_any_network_return_code, None
     new_authorized_networks_needed = not current_machine_already_authorized
 
-  if new_authorized_networks_needed:
-    return_code, authorized_networks = add_current_machine_to_networks(
-        authorized_networks
-    )
-    if return_code > 0:
-      xpk_print(
-          "Adding current machine's IP address to the authorized networks"
-          ' failed!'
-      )
-      return return_code
-
-    cluster_authorized_networks_update_code = (
-        update_cluster_authorized_networks(args, authorized_networks)
-    )
-    if cluster_authorized_networks_update_code > 0:
-      xpk_print('Updating cluster authorized networks failed!')
-      return cluster_authorized_networks_update_code
-    xpk_print("Cluster's master authorized networks updated successfully.")
-  else:
-    xpk_print('Current machine is already in authorized networks.')
-
-  return 0
+  return 0, authorized_networks if new_authorized_networks_needed else None
```

Comment:
I still don't like this piece of code as a whole - it's convoluted, some logic happens inside the get_authorized_networks_to_update function but still we have to handle the adding of the current network outside of it? My intention was - if this can't be any cleaner at least encapsulate it as a whole in a separate function. 

This should be a simple piece of logic:
- get networks if there are new, use the old ones otherwise
- check if the current network is in the desired networks, add it if not
- update or not the networks

Currently it doesn't really read like this. Can we divide the logic to these readable blocks?

Sorry about the nagging, I just had to think about this a whole lot just to understand it, and it shouldn't be this hard :(